### DISCUSSION TITLE
Two different export createBrowserRouter when Page is lazy route using recoil to update store will show fail message

### DISCUSSION THREAD

**Discussion by johnkao3** [remix-run/react-router#10829] [Q&A] [answered]
My example code is here [Example Repo](https://github.com/johnkao3/lazy-router-issue/blob/main/src/router.tsx) 

Package version is
- "react": "^18.2.0"
- "react-dom": "^18.2.0"
- "react-router-dom": "^6.15.0"
- "recoil": "^0.7.7"

This export no show warning message
<img width="555" alt="截圖 2023-08-29 下午4 30 26" src="https://github.com/remix-run/remix/assets/21374882/2fea5558-75b3-455b-b65d-82899431edf1">

But this one will show "Matched leaf route at location "/about" does not have an element or Component. This means it will render an <Outlet /> with a null value by default resulting in an "empty" page."
<img width="752" alt="截圖 2023-08-29 下午4 30 09" src="https://github.com/remix-run/remix/assets/21374882/3a284ad2-deb2-4f33-a6f7-6b31f219a70d">

Can anyone tell me the difference?

**Comment by brophdawg11** [comment, marked as answer]
You shouldn't be creating your data router inside of the react render lifecycle, otherwise it will be destroyed and recreated on each render.  Instead, it should be created as a singleton outside of React and passed to `RouterProvider`.  There's an example of this setup in the [docs](https://reactrouter.com/en/main/routers/create-browser-router).
